rcl/test/rcl/test_client.cpp

```diff
@@ -129,6 +129,21 @@ TEST_F(TestClientFixture, test_client_init_fini) {
   EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret) << rcl_get_error_string_safe();
   rcl_reset_error();
 
+  // Check if null publisher is valid
+  EXPECT_EQ(rcl_client_is_valid(nullptr), false);
+  rcl_reset_error();
+
+  // Check if zero initialized client is valid
+  client = rcl_get_zero_initialized_client();
+  EXPECT_EQ(rcl_client_is_valid(&client), false);
+  rcl_reset_error();
+
+  // Chek that a valid client is valid
+  client = rcl_get_zero_initialized_client();
+  ret = rcl_client_init(&client, this->node_ptr, ts, topic_name, &default_client_options);
```

Comment:
Would it make sense to check the value of ret before testing if it's valid? same below

Reply:
I was wondering about this myself, actually. I didn't see it as completely necessary, but I don't mind adding an `EXPECT_EQ(RCL_RET_OK, ret)` if requested. I'll defer to you.

Reply:
Given that every other test does it I think it makes sense to do it in the newly added tests as well.

Please also print the error in case it's not returning the right error code:
`EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();`